src/controllers/expensesControllers.js

```diff
@@ -0,0 +1,141 @@
+'use strict';
+
+const usersControllers = require('./usersControllers.js');
+
+let expenses = [];
+
+const init = () => {
+  expenses = [];
+};
+
+const getExpenses = (req, res) => {
+  const {
+    userId,
+    category,
+    from,
+    to,
+  } = req.query;
+
+  if (typeof +userId !== 'number') {
+    res.sendStatus(400);
+
+    return;
+  };
+
+  if (expenses.length === 0) {
+    res.send([]);
+
+    return;
+  };
+
+  if (usersControllers.findUser(+userId)) {
+    let userExpenses = expenses.filter(expense => expense.userId === +userId);
+
+    if (category) {
+      userExpenses = userExpenses.filter(
+        (expense) => expense.category === category
+      );
+    }
+    res.send(userExpenses);
+
+    return;
+  }
+
+  if (from && to) {
+    const foundExpenses = expenses.filter(
+      (expense) => expense.spentAt >= from && expense.spentAt <= to);
+
+    res.send(foundExpenses);
+
+    return;
+  }
+
+  res.send(expenses);
+};
+
+const createExpense = (req, res) => {
+  const {
+    userId,
+    spentAt,
+    title,
+    amount,
+    category,
+    note,
+  } = req.body;
```

Comment:
You could destructure only userId here
```suggestion
    const { userId } = req.body;
```
As I understand, you need only it. And in `expenseWitId` use `...req.body`.